src/tunnel/server/server.rs

```diff
@@ -89,68 +90,46 @@ impl WsServer {
             Pin<Box<dyn AsyncWrite + Send>>,
             bool,
         ),
-        Response<Either<String, BoxBody<Bytes, anyhow::Error>>>,
+        HttpResponse,
     > {
-        match extract_x_forwarded_for(req) {
-            Ok(Some((x_forward_for, x_forward_for_str))) => {
-                info!("Request X-Forwarded-For: {:?}", x_forward_for);
-                Span::current().record("forwarded_for", x_forward_for_str);
-                client_addr.set_ip(x_forward_for);
-            }
-            Ok(_) => {}
-            Err(_err) => return Err(bad_request()),
+        if let Some((x_forward_for, x_forward_for_str)) = extract_x_forwarded_for(req) {
+            info!("Request X-Forwarded-For: {x_forward_for:?}");
+            Span::current().record("forwarded_for", x_forward_for_str);
+            client_addr.set_ip(x_forward_for);
         };
 
-        let path_prefix = match extract_path_prefix(req) {
-            Ok(p) => p,
-            Err(_err) => return Err(bad_request()),
-        };
+        let path_prefix = extract_path_prefix(req)?;
 
         if let Some(restrict_path) = restrict_path_prefix {
             if path_prefix != restrict_path {
                 warn!(
-                    "Client requested upgrade path '{}' does not match upgrade path restriction '{}' (mTLS, etc.)",
-                    path_prefix, restrict_path
+                    "Client requested upgrade path '{path_prefix}' does not match upgrade path restriction '{restrict_path}' (mTLS, etc.)"
                 );
                 return Err(bad_request());
             }
         }
 
-        let jwt = match extract_tunnel_info(req) {
-            Ok(jwt) => jwt,
-            Err(_err) => return Err(bad_request()),
-        };
+        let jwt = extract_tunnel_info(req)?;
 
         Span::current().record("id", &jwt.claims.id);
         Span::current().record("remote", format!("{}:{}", jwt.claims.r, jwt.claims.rp));
-        let remote = match RemoteAddr::try_from(jwt.claims) {
-            Ok(remote) => remote,
-            Err(err) => {
-                warn!("Rejecting connection with bad tunnel info: {} {}", err, req.uri());
-                return Err(bad_request());
-            }
-        };
+        let remote = RemoteAddr::try_from(jwt.claims)
+            .inspect_err(|err| warn!("Rejecting connection with bad tunnel info: {err} {}", req.uri()))
```

Comment:
It is a matter of taste, but I don't really like inspect_err followed by map_err. It is less clear than the match, as it splits the error block in 2 for no reason.